client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx

```diff
@@ -217,13 +280,68 @@ const PlanOnlyThankYou = ( {
 					/>
 				}
 				footerDetails={ footerDetails }
+				upsellProps={ upsellProps }
 			/>
 		</>
 	);
 };
 
-export default connect( null, {
-	errorNotice,
-	removeNotice,
-	successNotice,
-} )( PlanOnlyThankYou );
+function getBusinessPlanSlugToUpgrade( currentPlanSlug: string ) {
+	const currentPlan = getPlan( currentPlanSlug );
+	const planKey = findFirstSimilarPlanKey( PLAN_BUSINESS, { term: currentPlan?.term } );
+
+	// It's just for TS, since .find() can return undefined
+	if ( ! planKey ) {
+		return;
+	}
+
+	return getPlan( planKey );
+}
+
+export default connect(
+	( state: IAppState, props: PlanOnlyProps ) => {
+		let promotePlanData;
+		if ( isPersonal( props.primaryPurchase ) || isPremium( props.primaryPurchase ) ) {
```

Comment:
Just a suggestion - we could invert that condition so we have less lines indented:

```suggestion
		if ( ! isPersonal( props.primaryPurchase ) && ! isPremium( props.primaryPurchase ) ) {
			return {};
		}
```

Reply:
I was thinking about it and decided not to do it, since - ATM we have only one condition inside this `connect`'s callback, but in the future we can add some other conditions / configurations and if we now go with this "less lines indented" approach - in the future we will need to rewrite this our existing condition to not return empty object. But with the current approach - this existing condition will be not necessary to touch.

For example:
```
if ( isPersonal( props.primaryPurchase ) || isPremium( props.primaryPurchase ) ) {
promotePlanData = {...}
}

if ( isFree( props.primaryPurchase )) {
free = {...}
}

return {
   promotePlanData,
   free,
};
```
So the current code is straightforward - we have 1 `return` statement and above we are filling in it.

Otherwise it will be a bit messy.

Was I clear? Did you get the point? :) If no, I will elaborate with more details :)

Reply:
Yes, you were clear 🙂. I generally recommend avoiding premature optimization but you know this code more than me, as well as future improvements, so I'll stick to your suggestion.

Reply:
Just a bit more thoughts out loud - if it's a function specific for one action (e.g. getBusinessPlanToUpgrade - https://github.com/Automattic/wp-calypso/pull/90284/files#diff-dcc2b24affae1f0b97d49d49a1880e37baa86575c85d01480588444ee7811430R73), then it would totally makes sense to return earlier. But in this case it's *common* callback for connect, where we take `store` and transform it to `props`, so the main distinguish for me - is it *common* place or specific for some action.